src/geometry/types.rs

```diff
@@ -122,57 +127,63 @@ impl Vector {
     }
 }
 
-// Rectangle /////////////////////////////////////////////////////////////////
-#[derive(PartialEq, Debug)]
+// Size //////////////////////////////////////////////////////////////////////
+
+#[derive(Clone, PartialEq, Debug)]
 pub struct Size {
     pub width: f32,
     pub height: f32,
 }
 
-#[derive(PartialEq, Debug)]
+impl Size {
+    pub fn new(width: f32, height: f32) -> Size {
+        if width <= 0.0 || height <= 0.0 {
+            panic!("A size's width and/or height must be positive.");
```

Comment:
so if the condition of the if condition is not valid, it will print the Sting in panic?

Reply:
If the width or height is zero or less, then the program will crash and print the message to the console.